File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] solution = new int[2];
        HashMap <Integer, Integer> array_values = new HashMap<>();
        for (int i = 0; i < nums.length; i++){
            if(array_values.containsKey(target - nums[i])){
                solution[0] = array_values.get(target - nums[i]);
                solution[1] = i; 
            } else {
                array_values.put(nums[i], i);
            }
        }
        return solution;
        
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l = 0
        r = len(numbers) - 1
        res = [0,0] 
        #if list is empty
        if len(numbers) == 0:
            return []
        while l < r:
            summation = numbers[l] + numbers[r]
            if summation == target:
                res[0] = l + 1
                res[1] = r + 1
                break
            elif summation < target:
                l+=1
            else:
                r -= 1
        return res


        